flash/audio/speech_recognition/data.py

```diff
@@ -0,0 +1,250 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from dataclasses import dataclass
+from functools import partial
+from typing import Any, Callable, Dict, List, Mapping, Optional, Sequence, Tuple, Union
+
+import soundfile as sf
+import torch
+from torch import Tensor
+
+from flash.core.data.auto_dataset import AutoDataset
+from flash.core.data.data_module import DataModule
+from flash.core.data.data_source import DataSource, DefaultDataSources
+from flash.core.data.process import Deserializer, Postprocess, Preprocess
+from flash.core.utilities.imports import _TEXT_AVAILABLE
+
+if _TEXT_AVAILABLE:
+    from datasets import load_dataset
+    from transformers import Wav2Vec2CTCTokenizer, Wav2Vec2Processor
+
+
+class AudioDeserializer(Deserializer):
+
+    def __init__(self, backbone: str, max_length: int):
+        super().__init__()
+        self.backbone = backbone
+        self.tokenizer = Wav2Vec2CTCTokenizer.from_pretrained(backbone)
+        self.max_length = max_length
+
+    def deserialize(self, sample: Any) -> Tensor:
+        return self.tokenizer(sample["speech"], sampling_rate=sample["sampling_rate"][0]).input_values
```

Comment:
This should be wrapped into a dictionary for consistency.
{DEFAULTKEYS.INPUT: ...}